### PR TITLE
238 pause transition

### PR DESCRIPTION
**Pull request by anncatton**
Creates endpoint for system level batch jobs. Only pausing is implemented for now.
- adds `attestedAtUtc` value for when user attests; used in pauseable apps query
- updates swagger
- adds daco system scope to config
- paused email not implemented
- makes attestation period of 1 year from approval configurable for testing in dev/qa
- adds special admin pause endpoint for testing, configurable for env
- new config values needed: `DACO_SYSTEM_SCOPE`, `ATTESTATION_UNIT_COUNT`, `ATTESTATION_UNIT_OF_TIME`, and `ADMIN_PAUSE`

### CODE REVIEW

File: src/domain/state.ts [hunk 1]
```diff
@@ -996,12 +1023,39 @@ function updateAppStateForApprovedApplication(
...
+    }
+    return transitionToPaused(currentApplication, updatedBy, updatePart.pauseReason);
+  }
   if (currentApplication.sections.ethicsLetter.declaredAsRequired && updateDocs) {
     delete updatePart.sections?.ethicsLetter?.declaredAsRequired;
     updateEthics(updatePart, currentApplication, updateDocs);
   }
 }
 
+function updateAppStateForPausedApplication(
```
Comment (anncatton) [inline src/domain/state.ts:1042]: Added this in this PR so I could transition PAUSED apps back to APPROVED for testing

File: src/domain/service.ts [hunk 2]
```diff
@@ -528,6 +534,101 @@ export const searchCollaboratorApplications = async (identity: Identity) => {
...
+  currentDate: Date,
+) => {
+  const pauseableApps = await searchPauseableApplications(currentDate);
+  const config = await getAppConfig();
+  if (pauseableApps.length === 0) {
+    logger.info('No applications need to be paused at this time.');
+    return report.pausedApps;
+  }
+
+  await Promise.all(
```
Comment (blabadi) [inline src/domain/service.ts:605]: if one promise fails the whole process fails check [Promise.allSettelled](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise/allSettled)

File: src/domain/service.ts [hunk 3]
```diff
@@ -528,6 +534,101 @@ export const searchCollaboratorApplications = async (identity: Identity) => {
...
+
+  // can still run this multiple times in the same 24hr period,
+  // as any apps that were paused previously will be ignored because we're looking for APPROVED state
+  const apps = await ApplicationModel.find({
+    state: 'APPROVED',
+    approvedAtUtc: {
+      $gte: approvalDayStart, // check for an approvedAtUtc within a 24hr period
+      $lte: approvalDayEnd,
+    },
+    attestedAtUtc: { $exists: false }, // check the applicant has not already attested
```
[the remark this answers is not in the document]
Comment (anncatton) [reply src/domain/service.ts:561]: should i make that value nullable? i'm wondering if it makes sense once we get to renewals and attestations need to be repeated...

File: src/domain/service.ts (hunk 2 above)
Comment (blabadi) [reply src/domain/service.ts:605]: you could also use `updateMany` which can bulk update stuff

File: src/domain/service.ts (hunk 3 above)
[the remark this answers is not in the document]
Comment (blabadi) [reply src/domain/service.ts:561]: yeah I think that makes sense

**Review by blabadi — APPROVED** [review]

**Pull request merged**